### PR TITLE
refactor: replace the web api with app api

### PR DESCRIPTION
**Pull request by HaitianLiu**

### CODE REVIEW

**Comment by HaitianLiu** [comment]
周日要加班，晚上花了些时间，跑了测试用例，草草提交了。 @Libitum 辛苦再review一下～

File: Jellyfin.Plugin.Douban/FrodoUtils.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+using Jellyfin.Plugin.Douban.Response;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Model.Entities;
+
+namespace Jellyfin.Plugin.Douban
+{
+    /// <summary>
+    /// Frodo is the secondary domain of API used by Douban APP.
+    /// </summary>
+    public class FrodoUtils
```
Comment (Libitum) [inline Jellyfin.Plugin.Douban/FrodoUtils.cs:16]: nit: Maybe we can treat this as the douban frodo client, then we can encapsulate more related logic here. WDYT?

**Review by Libitum — COMMENTED** [review]
Thanks for the code! Just a few questions to be discussed.

File: Jellyfin.Plugin.Douban/Configuration/PluginConfiguration.cs [hunk 2]
```diff
@@ -8,7 +8,6 @@ public class PluginConfiguration : BasePluginConfiguration
         public int MinRequestInternalMs { get; set; }
         public PluginConfiguration()
         {
-            ApiKey = "0df993c66c0c636e29ecbb5344252a4a";
             MinRequestInternalMs = 2000;
```
[the remark this answers is not in the document]
Comment (HaitianLiu) [reply Jellyfin.Plugin.Douban/Configuration/PluginConfiguration.cs:11]: No need. Client API has no request limit.
However, your original solution still needs it. Better to leave this.

File: Jellyfin.Plugin.Douban/BaseProvider.cs [hunk 3]
```diff
@@ -42,160 +40,115 @@ public abstract class BaseProvider
         }
 
         public Task<HttpResponseInfo> GetImageResponse(string url,
-            CancellationToken cancellationToken)
+           CancellationToken cancellationToken)
         {
-            _logger.LogInformation("Douban:GetImageResponse url: {0}", url);
+            _logger.LogInformation("[DOUBAN INFO] GetImageResponse url: {0}", url);
```
[the remark this answers is not in the document]
Comment (HaitianLiu) [reply Jellyfin.Plugin.Douban/BaseProvider.cs:45]: Got it. Better to seal it in _logger.LogInformation.

File: Jellyfin.Plugin.Douban/SearchCache.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using Jellyfin.Plugin.Douban.Response;
+
+namespace Jellyfin.Plugin.Douban
+{
+    public sealed class SearchCache
```
[the remark this answers is not in the document]
Comment (HaitianLiu) [reply Jellyfin.Plugin.Douban/SearchCache.cs:6]: It's the simplest implement, but useful. In most cases, the same request is next to each other.

File: Jellyfin.Plugin.Douban/FrodoUtils.cs (hunk 1 above)
Comment (HaitianLiu) [reply Jellyfin.Plugin.Douban/FrodoUtils.cs:16]: I think the next step is to package the frodo code into lib, same like [TMDbLib](https://github.com/LordMike/TMDbLib).

**Pull request merged**